manage_containers.py

```diff
@@ -0,0 +1,189 @@
+
+class Container:
+    """
+    Represents a Docker container configured for kernel building
+
+    Class Attributes:
+        sudo (str): represents sudo command
+        quiet (bool): enable quiet option in Docker
+
+    Container Attributes:
+        gcc (str): Version of GCC compiler
+        clang (str): Version of Clang compiler (optional)
+        ubuntu (str): Version of Ubuntu
+        id (str): Docker image id
+    """
+
+    sudo = ''
+    quiet = False
+
+    def __init__(self, gcc_version, clang_version, ubuntu_version):
+        self.gcc = gcc_version
+        self.clang = clang_version
+        self.ubuntu = ubuntu_version
+        self.id = self.check()
+
+    def add(self):
+        """Builds the Docker container with the specified compilers"""
+        build_args = ['--build-arg', f'GCC_VERSION={self.gcc}',
+                      '--build-arg', f'CLANG_VERSION={self.clang}',
+                      '--build-arg', f'UBUNTU_VERSION={self.ubuntu}',
+                      '--build-arg', f'UNAME={os.getlogin()}',
+                      '--build-arg', f'UID={os.getuid()}',
+                      '--build-arg', f'GID={os.getgid()}',
+                      '-t', f'kernel-build-container:gcc-{self.gcc}',
+                      '-t', f'kernel-build-container:clang-{self.clang}'
+        ]
+
+        if self.quiet:
+            build_args += ['-q']
+        subprocess.run([self.sudo, 'docker', 'build', *build_args, '.'],
+                        text=True, check=True)
+        self.check()
+
+    def rm(self):
+        """Removes the Docker container if it exists and not running"""
+        running = subprocess.run(f"{Container.sudo} docker ps | "
+                                 f"grep -E 'kernel-build-container:"
+                                 f"(gcc-{self.gcc}|clang-{self.clang})' || true",
+                                 shell=True, text=True, check=True, stdout=subprocess.PIPE).stdout
+        if not running:
+            subprocess.run([self.sudo, 'docker', 'rmi', '-f', self.id],
+                            text=True, check=True)
+            self.check()
+            return ''
+        return running
+
+    def check(self):
+        """Checks if the Docker container exists and parse id"""
+        cmd = subprocess.run([self.sudo, 'docker', 'images',
+                              f'kernel-build-container:clang-{self.clang}', '--format', '{{.ID}}'],
+                              stdout=subprocess.PIPE,
+                              text=True, check=True)
+        self.id = cmd.stdout.strip()
+        return self.id
+
+def check_group():
+    """Checks if the user is in the Docker group, returns 'sudo' if not"""
+    result = subprocess.run(['groups'], capture_output = True,
+                            text = True, check = True)
+    if 'docker' in result.stdout:
+        return ''
+    print('We need to use sudo for running docker')
+    return 'sudo'
+
+def add_handler(needed_compiler, containers):
```

Comment:
This function contains a lot of copy-pasted code.
Please refactor it. You can make it much simpler.